### ISSUE TITLE
Pretrain with categorical data

### ISSUE THREAD

**Issue by siyi-wind** [jsyoon0823/VIME#10]
Hi Jinsung. Thank you very much for introducing this wonderful work and for sharing the code! Would you mind telling me how do you do pretraining on categorical data. For example, after one-hot encoding, a sample x=[2, 0.9, -0.2] (say x[0] is a categorical data) would be [0, 0, 1, 0.9, -0.2]. In this case, how do you mask the data and reconstruct it?

Thank you very much for your support and consideration to this matter.

**Comment by jsyoon0823** [comment]
There are multiple ways of handling categorical features.

One way would be making multi-head architecture and using softmax
activation as the output for one-hot encoded categorical data.
For the rest of the continuous values, you can use another head with the
activation function in the github codebase.

Thanks!
